code/datums/wounds/burns.dm

```diff
@@ -70,42 +70,41 @@
 	// sanitization is checked after the clearing check but before the actual ill-effects, because we freeze the effects of infection while we have sanitization
 	if(sanitization > 0)
 		var/bandage_factor = (limb.current_gauze ? limb.current_gauze.splint_factor : 1)
-		infestation = max(0, infestation - WOUND_BURN_SANITIZATION_RATE)
-		sanitization = max(0, sanitization - (WOUND_BURN_SANITIZATION_RATE * bandage_factor))
+		infestation = max(infestation - (WOUND_BURN_SANITIZATION_RATE * delta_time), 0)
+		sanitization = max(sanitization - (WOUND_BURN_SANITIZATION_RATE * bandage_factor * delta_time), 0)
 		return
 
-	infestation += infestation_rate
-
+	infestation += infestation_rate * delta_time
 	switch(infestation)
 		if(0 to WOUND_INFECTION_MODERATE)
 		if(WOUND_INFECTION_MODERATE to WOUND_INFECTION_SEVERE)
-			if(prob(30))
+			if(DT_PROB(15, delta_time))
 				victim.adjustToxLoss(0.2)
 				if(prob(6))
 					to_chat(victim, "<span class='warning'>The blisters on your [limb.name] ooze a strange pus...</span>")
 		if(WOUND_INFECTION_SEVERE to WOUND_INFECTION_CRITICAL)
-			if(!disabling && prob(2))
+			if(!disabling && DT_PROB(1, delta_time))
 				to_chat(victim, "<span class='warning'><b>Your [limb.name] completely locks up, as you struggle for control against the infection!</b></span>")
 				set_disabling(TRUE)
-			else if(disabling && prob(8))
+			else if(disabling && DT_PROB(4, delta_time))
 				to_chat(victim, "<span class='notice'>You regain sensation in your [limb.name], but it's still in terrible shape!</span>")
 				set_disabling(FALSE)
-			else if(prob(20))
+			else if(DT_PROB(10, delta_time))
 				victim.adjustToxLoss(0.5)
 		if(WOUND_INFECTION_CRITICAL to WOUND_INFECTION_SEPTIC)
-			if(!disabling && prob(3))
+			if(!disabling && DT_PROB(1.5, delta_time))
 				to_chat(victim, "<span class='warning'><b>You suddenly lose all sensation of the festering infection in your [limb.name]!</b></span>")
 				set_disabling(TRUE)
-			else if(disabling && prob(3))
+			else if(disabling && DT_PROB(1.5, delta_time))
 				to_chat(victim, "<span class='notice'>You can barely feel your [limb.name] again, and you have to strain to retain motor control!</span>")
 				set_disabling(FALSE)
-			else if(prob(1))
+			else if(DT_PROB(0.5, delta_time))
 				to_chat(victim, "<span class='warning'>You contemplate life without your [limb.name]...</span>")
 				victim.adjustToxLoss(0.75)
-			else if(prob(4))
+			else if(DT_PROB(2, delta_time))
 				victim.adjustToxLoss(1)
```

Comment:
Is this actually the proper conversion for else ifs? Could be wrong, but if wait is raised the probabilities wouldn't stay the same, would they?

Reply:
You're right. I'll fix it.